s3transfer/download.py

```diff
@@ -131,6 +131,27 @@ def _get_temp_fileobj(self):
         return f
 
 
+class DownloadSeekableOutputManager(DownloadOutputManager):
+    @classmethod
+    def is_compatible(self, download_target):
+        return (
+            hasattr(download_target, 'seek') and
+            hasattr(download_target, 'tell')
```

Comment:
An `io.IOBase` will have `seek`, but is not necessarily be seek-able or tell-able. You need to call `io.IOBase.seekable` to be able to tell.